cli/doc/parser.rs

```diff
@@ -48,7 +57,23 @@ impl From<BufferedError> for Vec<Diagnostic> {
   }
 }
 
+pub trait DocFileLoader {
+  fn resolve(
+    &self,
+    specifier: &str,
+    referrer: &str,
+  ) -> Result<ModuleSpecifier, OpError> {
+    ModuleSpecifier::resolve_import(specifier, referrer).map_err(OpError::from)
+  }
+
+  fn load_source_code(
```

Comment:
output `SourceFile` and change SWC lexer settings based on media type (short circuit for binary files like `.wasm`)